GSA/openacr-editor
# Provide option to disable chapters/tables

Progress on https://github.com/GSA/openacr/issues/265

Summary of changes:
* Added `disabled` true/false attribute to evaluation structure for each chapter.
* Added "disabled" checkboxes for chapters/tables on the about page with some help text.
* Added the same checkbox to appear on each chapter/table page that is also sync'd.
* Disabling the fields (and removing the view on report link) on the chapter/table page whenever the box is checked. 
* Removing the table from the report and HTML whenever the box is checked.
* Removing the progress bar, adjusting count in the sidebar whenever the box is checked.
* Unchecking the box will re-enable the fields, table, progress bars.
* Added tests.

**QA new report**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. Switch to the about page.
4. Confirm you see a section at the bottom for 'Disabled Chapters/Tables'.
5. Confirm you are given checkboxes for each chapter and table.
6. Check some of the boxes and confirm in the sidebar the progress bars for the disabled sections are removed and the total criteria count is reduced.
7. Navigate to the disabled chapter/table.
8. Confirm at the top you see a checkbox under 'Notes' with the label 'Disabled?' and confirmed it is checked.
9. Expand all the sections.
10. Confirm the fields are disabled.
11. Uncheck the 'Disabled?' box and confirm the fields are re-enabled and the progress bars are back and total criteria count is increase in the sidebar.
12. Check the 'Disabled?' box again.
13. Click the 'View report' or 'Report' tab.
14. Confirm you see a 'Valid' message.
15. Confirm in the report the disabled chapter/table does not display a 'Criteria' table.
16. Download the YAML & HTML report and confirm the HTML report does not display a 'Criteria' table but the YAML does have the data in it.

**QA existing report**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. Click 'Open report'.
4. Upload a yaml file from OpenACR. Try the 'drupal-9.yaml' or any of the yaml files from https://github.com/GSA/openacr/tree/main/openacr.
5. Switch to the about page.
6. Confirm you see a section at the bottom for 'Disabled Chapters/Tables'.
7. Confirm you are given checkboxes for each chapter and table.
8. Check some of the boxes and confirm in the sidebar the progress bars for the disabled sections are removed and the total criteria count is reduced.
9. Navigate to the disabled chapter/table.
10. Confirm at the top you see a checkbox under 'Notes' with the label 'Disabled?' and confirmed it is checked.
11. Expand all the sections.
12. Confirm the fields are disabled.
13. Uncheck the 'Disabled?' box and confirm the fields are re-enabled and the progress bars are back and total criteria count is increase in the sidebar.
14. Check the 'Disabled?' box again.
15. Click the 'View report' or 'Report' tab.
16. Confirm you see a 'Valid' message.
17. Confirm in the report the disabled chapter/table does not display a 'Criteria' table.
18. Download the YAML & HTML report and confirm the HTML report does not display a 'Criteria' table but the YAML does have the data in it.
